pfcpiface/utils.go

```diff
@@ -65,10 +65,17 @@ func inc(ip net.IP) {
 }
 
 func ip2int(ip net.IP) uint32 {
+	// Guard against nil or empty IP to prevent out-of-bounds access
+	if len(ip) == 0 {
+		return 0
+	}
 	if len(ip) == 16 {
 		return binary.BigEndian.Uint32(ip[12:16])
 	}
-
+	// Guard against IP with insufficient length (need at least 4 bytes for uint32)
+	if len(ip) < 4 {
+		return 0
+	}
```

Comment:
Why do you need to compare `if len(ip) == 0` and `if len(ip) < 4`? Why not just doing the following?
```suggestion
	// Guard against IP with insufficient length (need at least 4 bytes for uint32)
	if len(ip) < 4 {
		return 0
	}
	if len(ip) == 16 {
		return binary.BigEndian.Uint32(ip[12:16])
	}
```